src/twisted/names/common.py

```diff
@@ -78,79 +84,160 @@ def query(self, query, timeout=None):
             return defer.maybeDeferred(method, query.name.name, timeout)
 
     def _lookup(self, name, cls, type, timeout):
+        """
+        Umbrella defer-as-failed logic.
+        """
```

Comment:
:+1:

Reply:
Thanks.

I will follow the inheritance and try to allocate my documentation input as appropriate.

Perhaps I should just leave a stub here pointing back to twisted.internet.interfaces.IResolver

What do you think?

Reply:
In the case of `_lookup` &mdash; being an implementation detail not part of `IResolver` &mdash; I think that the docstring should highlight that this is the method subclasses of `ResolverBase` are supposed to override. Perhaps the docstring of `ResolverBase` should also highlight this?